### PR TITLE
feat: Action redesign: Updating the config for Google AI plugin to use sections and zones format

### PR DESCRIPTION
**Pull request by ankitakinger**
## Description

Action redesign: Updating the config for Google AI plugin to use sections and zones format

Fixes [#35493](https://github.com/appsmithorg/appsmith/issues/35493)

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10700147940>
> Commit: dda8cee018765fc2827d3bfed214057dda73c692
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10700147940&attempt=2" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.All`
> Spec:
> <hr>Wed, 04 Sep 2024 13:18:18 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Enhanced chat interface with improved control organization through the introduction of `DOUBLE_COLUMN_ZONE` and `SINGLE_COLUMN_ZONE`.
	- Upgraded section functionality with `SECTION_V2` for better usability and clarity.
	- More structured layout for dropdown controls, allowing for easier navigation and interaction.

- **Bug Fixes**
	- Resolved layout issues by restructuring child elements for improved user experience.

### CODE REVIEW

File: app/server/appsmith-plugins/googleAiPlugin/src/main/resources/editor/generate.json [hunk 1]
```diff
@@ -1,77 +1,89 @@
...
-      "conditionals": {
-        "enable": "{{true}}",
-        "fetchDynamicValues": {
-          "condition": "{{actionConfiguration.formData.command.data === 'GENERATE_CONTENT'}}",
-          "config": {
-            "params": {
-              "requestType": "GENERATE_CONTENT_MODELS",
-              "displayType": "DROP_DOWN"
+      "controlType": "DOUBLE_COLUMN_ZONE",
+      "identifier": "CHAT-Z1",
```
Comment (alex-golovanov) [inline app/server/appsmith-plugins/googleAiPlugin/src/main/resources/editor/generate.json:10]: nit: it doesn't appear that you need to specify identifiers for zones.

**Review by alex-golovanov — APPROVED** [review]

**Comment by ankitakinger** [comment]
Testing completed on EE. Everything looks good.

File: app/server/appsmith-plugins/googleAiPlugin/src/main/resources/editor/generate.json (hunk 1 above)
Comment (ankitakinger) [reply app/server/appsmith-plugins/googleAiPlugin/src/main/resources/editor/generate.json:10]: I think we need to add identifiers. Correct me if I'm wrong @albinAppsmith?

Comment (albinAppsmith) [reply app/server/appsmith-plugins/googleAiPlugin/src/main/resources/editor/generate.json:10]: Identifiers are mandatory when we apply conditionals.

Comment (alex-golovanov) [reply app/server/appsmith-plugins/googleAiPlugin/src/main/resources/editor/generate.json:10]: But you're not applying condition to the zone itself. I mean, it is not referenced anywhere. 

@albinAppsmith Please clarify, in this particular case, what is the purpose of the referenced identifier for the zone? (`"identifier": "CHAT-Z1"`)

**Review by albinAppsmith — APPROVED** [review]

**Pull request merged**